chunk/chunk_cache.go

```diff
@@ -168,9 +177,9 @@ func (c *Cache) StoreChunkData(ctx context.Context, userID string, chunk *Chunk)
 func (c *Cache) StoreChunks(ctx context.Context, userID string, chunks []Chunk) error {
 	errs := make(chan error)
 	for _, chunk := range chunks {
-		go func(chunk *Chunk) {
+		go func(chunk Chunk) {
 			errs <- c.StoreChunkData(ctx, userID, chunk)
-		}(&chunk)
+		}(chunk)
 	}
 	var errOut error
 	for i := 0; i < len(chunks); i++ {
```

Comment:
Maybe not in this PR, but it would be nice to at least log other errors below. (Right now logged will be only last one)